File: Raven.Client.Java/src/main/java/net/ravendb/client/listeners/IDocumentConflictListener.java
package net.ravendb.client.listeners;

import java.util.List;

import net.ravendb.abstractions.basic.Reference;
import net.ravendb.abstractions.data.JsonDocument;


/**
 * Hooks for users that allows you to handle document replication conflicts
 */
public interface IDocumentConflictListener {

  public boolean tryResolveConflict(String key, List<JsonDocument> results, Reference<JsonDocument> resolvedDocument);
}

File: src/Raven.NewClient/Listeners/IDocumentConflictListener.cs
using Raven.NewClient.Abstractions.Data;
using Raven.NewClient.Client.Data;

namespace Raven.NewClient.Client.Listeners
{
    /// <summary>
    /// Hooks for users that allows you to handle document replication conflicts
    /// </summary>
    public interface IDocumentConflictListener
    {
        bool TryResolveConflict(string key, JsonDocument[] conflictedDocs, out JsonDocument resolvedDocument);
    }
}
